### PR TITLE
Consolidate documentation in spk code base

### LINKED ISSUE #735: Documentation on `spk infra` implementation
This is supplementary to https://github.com/yradsmikham/spk-infra/blob/master/README.md. 

The goal is go a little deeper into how `spk infra` will handle specific situations in scaffolding and generation.

### PR DESCRIPTION
**Pull request by yradsmikham**
Closes https://github.com/microsoft/bedrock/issues/735

### CODE REVIEW

**Review by andrebriggs — CHANGES_REQUESTED** [review]
It would be good to have a table of contents for the infra section such that docs like `spk-terragrunt-infra.md` are discoverable. Perhaps at `docs/infra/README.md` we have a bulleted list of all the docs in the `docs/infra/*` directory

**Comment by andrebriggs** [comment]
The only other thing I would suggest is add a reference to `docs/infra/README.md` from `docs/cloud-infra-management.md`. Otherwise looks good

**Review by andrebriggs — APPROVED** [review]

File: docs/infra/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# SPK Infrastructure Design Documentations
+
+There has been several design iterations for Cluster/Infrastructure creation and
```
Comment (edaena) [inline docs/infra/README.md]: nit: is it "have" been instead of has been?

**Review by samiyaakhtar — APPROVED** [review]
That's a lot of great documentation, thanks for writing it 👍

**Review by edaena — APPROVED** [review]

**Pull request merged**